Fixing the installer to avoid permission issues

lhriley opened the pull request:
This fixes the installation issue here: https://github.com/luizm/asdf-shellcheck/issues/5

I also did some housekeeping to cleanup unused variables, and no longer write any files to disk during the installation process.

```
$ asdf plugin test shellcheck https://github.com/lhriley/asdf-shellcheck.git --asdf-plugin-gitref fix-installer shellcheck
Updating shellcheck to fix-installer
From https://github.com/lhriley/asdf-shellcheck
 * [new branch]      fix-installer -> fix-installer
Switched to branch 'fix-installer'
mkdir: created directory '/tmp/asdf.vVR2/installs/shellcheck/0.8.0/bin'
Downloading shellcheck from https://github.com/koalaman/shellcheck/releases/download/v0.8.0/shellcheck-v0.8.0.linux.x86_64.tar.xz to /tmp/asdf.vVR2/installs/shellcheck/0.8.0/bin
shellcheck-v0.8.0/shellcheck
ShellCheck - shell script analysis tool
version: 0.8.0
license: GNU General Public License, version 3
website: https://www.shellcheck.net
```

lhriley committed:
Fixing the installer to avoid permission issues

luizm commented on bin/install:
```diff
@@ -38,23 +37,24 @@ _get_download_url() {
+
+	echo "Downloading shellcheck from ${download_url} to ${bin_install_path}"
+	# curl to tar, without files on disk
+	( curl -Ls "${download_url}" | tar -xJv --strip-components=1 -C "${bin_install_path}" "shellcheck-v${version}/shellcheck" )
```
why is necessary subshell here?

lhriley replied:
It probably isn't necessary for functionality, and should work the same with or without it.

lhriley replied:
Removed in latest commit.

luizm commented on bin/install:
```diff
@@ -38,23 +37,24 @@ _get_download_url() {
+
+	local -r download_url="$(_get_download_url "${version}" "${platform}" "${arch}")"
+
+	mkdir -pv "${bin_install_path}"
```
Is necessary the mkdir output?

lhriley replied:
No, it looks like I left it in by accident. I used it for debugging.

lhriley replied:
Removed in latest commit.

luizm commented:
Thank you this PR @lhriley :) I added some comments.

lhriley committed:
Changes responding to comments

lhriley commented:
Updated output: 

```
$ asdf plugin test shellcheck https://github.com/lhriley/asdf-shellcheck.git --asdf-plugin-gitref fix-installer shellcheck --version
Updating shellcheck to fix-installer
From https://github.com/lhriley/asdf-shellcheck
 * [new branch]      fix-installer -> fix-installer
Switched to branch 'fix-installer'
shellcheck/bin/list-all is using GitHub API, just be sure you provide an API Authorization token
via your CI env GITHUB_API_TOKEN. This is the current rate_limit:
[...]
Downloading shellcheck from https://github.com/koalaman/shellcheck/releases/download/v0.8.0/shellcheck-v0.8.0.linux.x86_64.tar.xz to /tmp/asdf.uLUf/installs/shellcheck/0.8.0/bin
shellcheck-v0.8.0/shellcheck
ShellCheck - shell script analysis tool
version: 0.8.0
license: GNU General Public License, version 3
website: https://www.shellcheck.net

```

lhriley commented:
Thanks @luizm
